_posts/2020-05-29-Sourcify-Towards-Safer-Contract-Interaction-for-Humans.md

```diff
@@ -0,0 +1,120 @@
+date: '2020-05-29'
+author: Edi Sinovčić, Franziska Heintel
+category: Announcements
+---
+
+**tl;dr**: Building sensible blockchain applications for humans is **hard**. You can enhance the user experience of 
+your dapp today by leveraging the power of open source. Increase awareness and give more transparency on what users 
+are *actually* doing when interacting with your code on the blockchain, i.e. when signing a transaction, by publishing 
+the source code to this decentralized repository and using metadata files, which translate “random” hex strings into 
+human-readable language. Sourcify is a tool to help you do exactly that. If you want to upload your source code, either 
+visit verification.komputing.org, the [Sourcify Github repo](https://github.com/ethereum/sourcify) or use the Sourcify 
+plugin in Remix. To learn how Sourcify works and why it matters, read the whole blog post.
+
+You might already know Sourcify under the name “Source-Verify”/ ”Source-Verifier” - don’t despair, we just decided to give 
+it a nicer name!
+
+## YOLO Signing - One of Today’s Everyday Problems in Ethereum UX
+Think about the following: How many developers of the dapps you are using do you know personally?
+
+Let’s imagine you want to interact with a dapp using MetaMask and it shows you this dialogue: 
+
+![](https://i.imgur.com/CjzXm9p.png)
+
+*You Only Live Once (YOLO)  Signing - confirm at your peril*
+
+What does all that hex data mean? You basically have no idea what you are agreeing to  - but you only live once, right? 
+Do you reject or confirm? If you go ahead and sign, you are **YOLO signing**. YOLO signing is bad because even if you only 
+live once, you can probably regret some stuff forever! ;) 
+
+By confirming, you blindly trust the developers of this dapp that it will do what they promised. And while it’s great to 
+have faith in humanity, it’s even greater to have the option to verify what is happening and take an informed decision.
+
+## Source Verification & NatSpec to the Rescue
+
+To prevent your users from having to blindly trust you and YOLO sign, you, as a dapp developer, can verify your code.
+
+By verifying and uploading your code, you allow any other developer to recompile the source code and check it for themselves.
+
+Properly “verifying” your code involves two steps:
+1. Commenting the code using [NatSpec](https://solidity.readthedocs.io/en/latest/natspec-format.html). NatSpec is the 
+Ethereum Natural Language Specification Format which describes functions, return values etc. in a friendly way.
+1. Publishing the dapp’s source code and its metadata to a decentralized network.
+
+## Meet Sourcify, a decentralized automated contract verifying service
+
+Sourcify tries to make the process of source verification (step 2 in the list above) easy for you. There are different ways 
+how you can make use of the tool. 
+
+You can either...
+* Use the interface hosted on verification.komputing.org
```

Comment:
```suggestion
* Use the interface hosted on [verification.komputing.org](https://verification.komputing.org/)
```

Should also really consider having a sourcify domain :)

Reply:
yeah totally...